### ISSUE TITLE
Support LedgerX Futures, Options, and Spot trades

### ISSUE THREAD

**Issue by bplunkert** [rotki/rotki#4639]
## Abstract

This is part of the general Futures and Options support that Rotki needs. Same applies for other exchanges that have options and futures.

Related to #1606

## Motivation

Options trade support is a feature missing from all commercial SaaS competitors to this project. This would be a major differentiator for Rotki which would set it lightyears ahead of the competition.

## Specification

I am not sufficiently knowledgable about either Rotki or accounting to write a detailed spec, but broadly speaking a few areas could be covered:
* Importing LedgerX statements - there are at least 3-4 different CSV formats provided, and no one file contains sufficient information alone.
* General tax form reconciliation - These derivatives exchanges typically provide tax forms, which are normally incorrect as they do not include cost basis. Reconciling these forms is very difficult or impossible for most humans.
* Futures trading
* * PNL for futures positions closed before settlement
* * Delivery/settlement
* Options trading
* * PNL for options positions closed before settlement without delivery
* * Options exercise (resulting in disposition or acquisition of underlying assets, affecting overall cost basis for the underlying asset)
* * Options assignment (resulting in disposition or acquisition of underlying assets, affecting overall cost basis for the underlying asset)
* * Options expiration

**Comment by LefterisJP** [comment]
I am not a futures trader, but they are different to margin trades right? We have a generic issue for margin trades support: https://github.com/rotki/rotki/issues/1980

We don't have one yet for generic futures support. Perhaps this one could be it?

In order to properly implement either for any supported exchange we would need to have accounting support for those types of trades. And then for each exchange/location/import recognize them.

No idea on an ETA for any of this as our current focus is much more on on-chain understanding of transactions and accounting for the gazillion of chains that exist. And we are a small opensource project. But maybe at some point we can focus on it. Also if anyone from the community really wants support for either PRs are welcome!

**Comment by bplunkert** [comment]
> I am not a futures trader, but they are different to margin trades right? We have a generic issue for margin trades support: #1980

Yes, they are different. Some futures trading may be on margin, some is not. In the case of LedgerX there is no margin allowed, so all futures and options trading is 100% collateralized. I would keep futures and options separate from margin when planning the features implementation.

> We don't have one yet for generic futures support. Perhaps this one could be it?

I am not very familiar with the project, so I don't have a strong opinion, but as a trader the feature most important to me is options, which will be quite different to implement from futures. Due to the way LedgerX works, it is pretty much impossible to trade options there without also trading futures. So I think it would make sense to implement futures first... but I am not sure how widely futures accounting might have to vary due to differences in contracts/laws/etc.

> In order to properly implement either for any supported exchange we would need to have accounting support for those types of trades. And then for each exchange/location/import recognize them.
> 
> No idea on an ETA for any of this as our current focus is much more on on-chain understanding of transactions and accounting for the gazillion of chains that exist. And we are a small opensource project. But maybe at some point we can focus on it. Also if anyone from the community really wants support for either PRs are welcome!
Makes sense. I'd love to make a donation if I could condition it upon specific feature implementations. Please consider allowing earmarking of funds and dispersal conditional upon successful implementation. If that's already possible I apologize and please direct me where to send some money!
